openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,864 @@
-var tempWeight = {};
-var tempCatWeight = {};
-var pdTempPrimaryIndicator = [];
-var pdTempPrimaryIndicatorLevels = [];
-var pdTempPILevel = {};
-var pdTempCategoryIndicator = [];
-var parentChildKey = {};
-var sviParentChildKey = {};
-var pdTempCatWeight = {};
-var tempParentChildKey = [];
-var catIndicator = {};
-var tempCatSearchElements = [];
-var tempSviSearchElements = [];
-var tempSviWeight = '';
-var tempAalWeight = '';
-var tempAalValue = '';
-var tempIriWeight = '';
-var municipality = [];
-var outlierBreakPoint = 0.75; // used for the primary indicator outlier
-
-// Keep track of the utfGrid that has been selected last
-var selectedGrid;
+var layerAttributes;
+var sessionProjectDef = [];
+var selectedRegion;
+var sessionProjectDefStr;
+var projectLayerAttributes;
+var region = [];
+var districts = [];
 var baseMapUrl = new L.TileLayer('http://otile1.mqcdn.com/tiles/1.0.0/map/{z}/{x}/{y}.png');
-
 var app = new OQLeaflet.OQLeafletApp(baseMapUrl);
-
-var startApp = function() {
-    map = new L.Map('map', {
-        minZoom: 2,
-        scrollWheelZoom: false,
-        attributionControl: false,
-        maxBounds: new L.LatLngBounds(new L.LatLng(-90, -180), new L.LatLng(90, 180)),
-    });
-    map.setView(new L.LatLng(10, -10), 2).addLayer(baseMapUrl);
-
-    layers = {};
-
-    layerControl = L.control.layers(app.baseLayers);
-    map.panTo(new L.LatLng(39.399, -8.22));
-    map.setZoom(6);
-
-    // Duplicate layer warnning message
-    function showDuplicateMsg() {
-        $('#warning-duplicate').dialog('open');
+var indicatorChildrenKey = [];
+
+function createIndexSimple(la, index) {
+    var indicator = [];
+    // setup the indicator with all the municipalities
+    for (var ia = 0; ia < la.length; ia++) {
+        var temp = {};
+        temp.region = la[ia].properties[selectedRegion];
+        indicator.push(temp);
+        // districts is used inside of the d3 charts
+        // TODO change this to region
+        districts.push(la[ia].properties[selectedRegion]);
     }
-
-    $(document).ready(function() {
-        $('#warning-duplicate').dialog({
-            autoOpen: false,
-            hieght: 300,
-            width: 350,
-            modal: true
-        });
-    });
-
-    // Slider
-    $(function() {
-        $( '#slider-vertical' ).slider({
-            orientation: 'vertical',
-            range: 'min',
-            min: 0,
-            max: 100,
-            value: 16.666,
-            slide: function( event, ui ) {
-                $( '#econ-weight' ).val( ui.value );
-            }
-        });
-        $( '#econ-weight' ).val( $( '#slider-vertical' ).slider( 'value' ) );
-    });
-
-    // No Layer to remove warnning message
-    function showRemoveMsg() {
-        $('#warning-no-layer').dialog('open');
+    // Get the indicators children keys
+    for (var q = 0; q < index.length; q++) {
+        indicatorChildrenKey.push(index[q].name);
```

Comment:
Please be aware that the risk and primary indicators have field elements, while the theme's do not, this is why you will find in some cases the use of 'field' and in others 'name'